test/core/iomgr/resolve_address_test.cc

```diff
@@ -289,11 +375,22 @@ int main(int argc, char** argv) {
       // these unit tests under c-ares risks flakiness.
       test_invalid_ip_addresses();
       test_unparseable_hostports();
+    } else {
+      test_localhost_result_has_ipv6_first();
     }
     grpc_executor_shutdown();
   }
   gpr_cmdline_destroy(cl);
-
   grpc_shutdown();
```

Comment:
Doesn't this new test need to run before `grpc_shutdown()`?

Reply:
It doesn't need to in this case because in the last test, I added a `grpc_init()` and `grpc_shutdown()` around it.

The reason for this is just because `address_sorting_override_source_addr_factory_for_testing` works on a once-per-grpc-init basis... so IMO it's simplest if tests that use `address_sorting_override_source_addr_factory_for_testing` have one corresponding `grpc_init`/`grpc_shutdown` pair.

Alternately all of the tests in `resolve_address_test` could perhaps run within the same `grpc_init`/`grpc_shutdown` pair, but then we'd probably need some comment, e.g.:

```
// Do not add any tests below test_localhost_result_has_ipv4_first_when_ipv6_isnt_available
```

Reply:
This makes sense.  But I suggest adding a comment about it to clarify why it's done this way.

Reply:
Done, added a comment